nav2_costmap_2d/include/nav2_costmap_2d/inflation_layer.hpp

```diff
@@ -148,16 +144,16 @@ class InflationLayer : public Layer
    * @param src_y The y coordinate of the source cell
    * @return
    */
-  inline unsigned char costLookup(int mx, int my, int src_x, int src_y)
+  inline unsigned char costLookup(
+    unsigned int mx, unsigned int my, unsigned int src_x,
+    unsigned int src_y)
   {
-    unsigned int dx = abs(mx - src_x);
-    unsigned int dy = abs(my - src_y);
-    return cached_costs_[dx][dy];
+    unsigned int dx = (mx > src_x) ? mx - src_x : src_x - mx;
```

Comment:
This is nearly identical to the implementation of abs (https://code.woboq.org/userspace/glibc/stdlib/abs.c.html), please revert for readability.